spark/agent/src/main/scala/za/co/absa/spline/harvester/conf/DefaultSplineConfigurer.scala

```diff
@@ -65,11 +79,34 @@ class DefaultSplineConfigurer(configuration: Configuration, sparkSession: SparkS
     }
   }
 
-  override lazy val lineageDispatcher = HttpLineageDispatcher(configuration)
+  override lazy val lineageDispatcher: LineageDispatcher = {
+    configuration.getString(LINEAGE_DISPATCHER_CLASS, LINEAGE_DISPATCHER_CLASS_DEFAULT) match {
+      case LINEAGE_DISPATCHER_CLASS_DEFAULT => HttpLineageDispatcher(configuration)
+      case className =>
+        log debug s"Instantiating a lineage dispatcher for class name: $className"
+        Class.forName(className.trim)
+          .getConstructor(classOf[Configuration])
+          .newInstance(configuration)
+          .asInstanceOf[LineageDispatcher]
+    }
+  }
+
+  override lazy val attributeLineageExtractor: AbstractAttributeLineageExtractor = {
+    configuration.getString(LINEAGE_EXCTRACTOR_CLASS, LINEAGE_EXCTRACTOR_CLASS_DEFAULT) match {
+      case LINEAGE_EXCTRACTOR_CLASS_DEFAULT => new DisabledAttributeLineageExtractor
+      case className =>
+        log debug s"Instantiating an AttributeLineageExtractor for class name: $className"
+        Class.forName(className.trim)
+          .getConstructor()
+          .newInstance()
+          .asInstanceOf[AbstractAttributeLineageExtractor]
+    }
+  }
 
   private lazy val harvesterFactory = new LineageHarvesterFactory(
```

Comment:
alternatively if `harvesterFactory` or `harvester` class was configurable, we could remove explicit attribute lineage, and allow customizations to be added by creating an extended LineageHarvester(Factory) which wraps existing `spline` code. 
looks like the cleanest way to go for now...
@wajda